File: C++/struk_belanja.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    string nama, nim, hargaText;
    int harga, jumlah;

    cout << "Nama Barang   : ";
    getline(cin, nama);

    cout << "Harga (Rp)    : ";
    getline(cin, hargaText);

    // Hapus "Rp" dan titik dari harga input
    hargaText.erase(remove(hargaText.begin(), hargaText.end(), 'R'), hargaText.end());
    hargaText.erase(remove(hargaText.begin(), hargaText.end(), 'p'), hargaText.end());
    hargaText.erase(remove(hargaText.begin(), hargaText.end(), '.'), hargaText.end());

    harga = stoi(hargaText);

    cout << "Jumlah Barang : ";
    cin >> jumlah;
    cin.ignore();

    cout << "NIM           : ";
    getline(cin, nim);

    int tigaAkhir = stoi(nim.substr(nim.length() - 3));
    int satuAkhir = nim[nim.length() - 1] - '0';

    int minimal = tigaAkhir * 100;
    int total = harga * jumlah;
    int diskonPersen = (total >= minimal) ? satuAkhir : 0;
    double diskonRp = total * (diskonPersen / 100.0);
    double bayar = total - diskonRp;

    cout << "\n===== STRUK BELANJA =====\n";
    cout << "Nama Barang      : " << nama << endl;
    cout << "Harga Satuan     : Rp " << harga << endl;
    cout << "Jumlah Beli      : " << jumlah << endl;
    cout << "Total Harga      : Rp " << total << endl;
    cout << "Minimal Diskon   : Rp " << minimal << endl;
    cout << "Diskon (%)       : " << diskonPersen << "%\n";
    cout << "Diskon (Rp)      : Rp " << (int)diskonRp << endl;
    cout << "Total Bayar      : Rp " << (int)bayar << endl;
    cout << "=========================\n";

    return 0;
}

File: Python/struk_belanja.py
nama = input("Nama Barang   : ")
harga_text = input("Harga satuan (Rp) : ").strip().replace("Rp", "").replace(".", "")
harga = int(harga_text)
jumlah = int(input("Jumlah Barang : "))
nim = input("3 angka NIM terakhir           : ").strip()

tiga_akhir = int(nim[-3:])
satu_akhir = int(nim[-1])

minimal_pembelian = tiga_akhir * 100
total_harga = harga * jumlah
diskon_persen = satu_akhir if total_harga >= minimal_pembelian else 0
diskon_rp = total_harga * (diskon_persen / 100)
total_bayar = total_harga - diskon_rp

print("\n===== STRUK BELANJA =====")
print(f"Nama Barang      : {nama}")
print(f"Harga Satuan     : Rp {harga:,}".replace(",", "."))
print(f"Jumlah Beli      : {jumlah}")
print(f"Total Harga      : Rp {total_harga:,}".replace(",", "."))
print(f"Minimal Diskon   : Rp {minimal_pembelian:,}".replace(",", "."))
print(f"Diskon (%)       : {diskon_persen}%")
print(f"Diskon (Rp)      : Rp {int(diskon_rp):,}".replace(",", "."))
print(f"Total Bayar      : Rp {int(total_bayar):,}".replace(",", "."))
print("=========================")
